### PR TITLE
interp: fix append with 1 argument

### LINKED ISSUE #1224: Builtin append with only one argument fails
### The following program `sample.go` triggers an unexpected result

```go
package main

func main() {
	var C []int
	_ = append(C)
}
```

### Expected result

```console
$ go run ./sample.go
<no error>
```

### Got

```console
$ yaegi ./sample.go
run: ./main.go:5:6: CFG post-order panic: runtime error: index out of range [1] with length 1
```

### Yaegi Version

v0.9.23

### Additional Notes

In addition to the above error `append(nil)` fails with a nil-dereference error where the go compiler gives a suitable compiler error.

### PR DESCRIPTION
**Pull request by mvertes**
Fixes #1224.

### CODE REVIEW

File: interp/interp_eval_test.go [hunk 1]
```diff
@@ -495,6 +495,7 @@ func TestEvalSliceExpression(t *testing.T) {
 		{src: `a := []int{0,1,2,3}[1::4]`, err: "1:49: 2nd index required in 3-index slice"},
 		{src: `a := []int{0,1,2,3}[1:3:]`, err: "1:51: 3rd index required in 3-index slice"},
 		{src: `a := []int{0,1,2}[3:1]`, err: "invalid index values, must be low <= high <= max"},
+		{src: `a := []int{0,1,2}; append(a)`, res: "[0 1 2]"},
```
Comment (nrwiersma) [inline interp/interp_eval_test.go:498]: perhaps this can move to `TestEvalBuiltin` with the other `append` tests. We should also add a test for `append(nil)`, WDYT?

**Review by nrwiersma — CHANGES_REQUESTED** [review]

File: interp/interp_eval_test.go (hunk 1 above)
Comment (mvertes) [reply interp/interp_eval_test.go:498]: yes, done

**Review by nrwiersma — APPROVED** [review]
LGTM
